orangewidget/settings.py

```diff
@@ -795,9 +790,15 @@ def update_defaults(self, widget):
         self.settings_from_widget(widget)
         globs = self.global_contexts
         assert widget.context_settings is not globs
-        ids = {id(c) for c in globs}
-        globs += (c for c in widget.context_settings if id(c) not in ids)
-        globs.sort(key=lambda c: -c.time)
+        new_contexts = []
+        for context in widget.context_settings:
+            context = copy.deepcopy(context)
+            for setting, data, _ in self.provider.traverse_settings(data=context.values):
+                if setting.schema_only and hasattr(data, setting.name):
+                    delattr(data, setting.name)
+            if context not in globs:
```

Comment:
Context used to be matched by their id's to prevent multiplying the same context. This is no longer possible because we must copy the widget's context before removing schema-only settings and merge them into global contexts.

I hope `context not in globs` will indeed work. This also required removing the `time` attribute. I shed no tears for it; I have already considered eliminating it for a long time.